### DISCUSSION TITLE
What the contition of a puch with a tocken to trigger a workflow

### DISCUSSION THREAD

**Discussion by sbrunner** [community/community#27013] [Actions]
<p>In practical why this commit<br>
<a href="https://github.com/camptocamp/demo_geomapfish/commit/0f56604d77d7f1e370e1fc7ba45309030784d8b8" rel="nofollow noopener">https://github.com/camptocamp/demo_geomapfish/commit/0f56604d77d7f1e370e1fc7ba45309030784d8b8</a></p>
<p>didn’t trigger this workflow</p>
<p><a href="https://github.com/camptocamp/demo_geomapfish/blob/0f56604d77d7f1e370e1fc7ba45309030784d8b8/.github/workflows/ci.yaml" rel="nofollow noopener">https://github.com/camptocamp/demo_geomapfish/blob/0f56604d77d7f1e370e1fc7ba45309030784d8b8/.github/workflows/ci.yaml</a></p>
<p>?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/sbrunner">@sbrunner</a> ,</p>
<p>In your workflow (<a href="https://github.com/camptocamp/demo_geomapfish/blob/prod-2-5/.github/workflows/upgrade-25.yaml" rel="nofollow noopener"><strong>Upgrade 2.5</strong></a>), I noticed that you configure the input parameter on the <strong>checkout</strong> action like as below:</p>
<pre><code>- uses: actions/checkout@v2
  with:
    ref: 'prod-2-5'
</code></pre>
<p>When using the <strong>checkout</strong> action, few points you need to understand:</p>
<p><strong>1)</strong> If you do not specify the value of the input parameter " <strong>token</strong>", by default the action will use <strong>${{ github.token }}</strong> to set the value.</p>
<p><strong>2)</strong> The input parameter " <strong>persist-credentials</strong>" is used to define whether to configure the <strong>token</strong> or <strong>SSH</strong> key with the local <strong>git config</strong>. By default it is " <strong>true</strong>".</p>
<p><strong>3)</strong> When " <strong>persist-credentials</strong>" is " <strong>true</strong>", the <strong>checkout</strong> action will use the value of " <strong>token</strong>" or " <strong>ssh-key</strong>" to configure the <strong>credentials</strong> of the local <strong>git config</strong>. The configured <strong>credentials</strong> will be kept to continue to use in the subsequent steps in the same job, even if you provide a new ssh-key/token in the subsequent steps. The newly provided ssh-key/token will be ignore.</p>
<p><strong>4)</strong> When you use the repository’s <strong>GITHUB_TOKEN</strong> ( <strong>github.token</strong> ) to perform tasks on behalf of the GitHub Actions app, events triggered by the <strong>GITHUB_TOKEN</strong> will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.</p>
<p>In summary, there are two solutions to solve the problem you are facing:</p>
<p><strong>Solution 1:</strong></p>
<pre><code>- name: Checkout
  uses: actions/checkout@v2
  with:
    token: ${{ secrets.GITHUB_GOPASS_CI_TOKEN }}
. . .
- name: Push changes
  run: |
    git config user.email "&lt;ci@camptocamp.com&gt;"
    git config user.name "Continuous integration"
    . . .
    git push
</code></pre>
<p>Using your GitHub PAT to authenticate on the  <strong>checkout</strong> action, and let " <strong>persist-credentials</strong>" be " <strong>true</strong>". Because the PAT you provide has been configured as the <strong>credentials</strong> of the local <strong>git config</strong> , you do not need to provide the <strong>credentials</strong> when executing  <strong>git push</strong>.</p>
<p><strong>Solution 2:</strong></p>
<pre><code>- name: Checkout
  uses: actions/checkout@v2
  with:
    persist-credentials: false
. . .
- name: Push changes
  run: |
    git config user.email "&lt;ci@camptocamp.com&gt;"
    git config user.name "Continuous integration"
    . . .
    git push https://c2c-bot-gis-ci:${{ secrets.GITHUB_GOPASS_CI_TOKEN }}@github.com/camptocamp/demo_geomapfish.git
</code></pre>
<p>Set " <strong>persist-credentials</strong>" to be " <strong>false</strong>". Because no  <strong>credentials</strong>  is configured for the local <strong>git config</strong> , you need to provide the <strong>credentials</strong> when executing  <strong>git push</strong>.</p>
<p>Related docs:</p>
<p><a href="https://github.com/actions/checkout#usage" rel="nofollow noopener"><strong>Checkout V2</strong></a><br>
<a href="https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#using-the-github_token-in-a-workflow" rel="nofollow noopener"><strong>Authenticating with the GITHUB_TOKEN</strong></a></p>

**Comment by sbrunner** [comment]
<p>I tested the solution 1 and it’s working thanks :slight_smile:</p>
